File: Leetcode/383-Ransom-Note/RandomNote.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        if (ransomNote.length() > magazine.length()) {
            return false;
        }
        int count[26] = {0};
        for (int i=0; i<magazine.length(); i++) {
            int index = magazine.at(i)-'a';
            count[index]++;
        }
        for (int i=0; i<ransomNote.length(); i++) {
            int index = ransomNote.at(i)-'a';
            count[index]--;
            if (count[index]<0){
                return false;
            }
        }
        return true;
    }
};

File: Leetcode/383-Ransom-Note/RandomNote.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        if len(ransomNote) > len(magazine):
            return False

        count = [0]*26
        for c in magazine:
            count[ord(c)-97]+=1

        for c in ransomNote:
            index = ord(c)-97
            count[index]-=1
            if count[index]<0:
                return False
        return True
